Add docs for CA1871 and CA2264 (#39661)
dotnet/docs: docs/fundamentals/code-analysis/quality-rules/ca1871.md

---
title: "CA1871: Do not pass a nullable struct to 'ArgumentNullException.ThrowIfNull'"
description: "Learn about code analyzer rule CA1871 - Do not pass a nullable struct to 'ArgumentNullException.ThrowIfNull'"
ms.date: 02/25/2024
f1_keywords:

>  - CA1870

Reviewer: ```suggestion
 - CA1871
```

 - DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNull
helpviewer_keywords:
 - CA1871
dev_langs:
 - CSharp

>  - VB

Reviewer: ```suggestion
```

author: CollinAlpert
---

# CA1871: Do not pass a nullable struct to 'ArgumentNullException.ThrowIfNull'

| Property                            | Value                                                                |
|-------------------------------------|----------------------------------------------------------------------|
| **Rule ID**                         | CA1871                                                               |
| **Title**                           | Do not pass a nullable struct to 'ArgumentNullException.ThrowIfNull' |
| **Category**                        | [Performance](performance-warnings.md)                               |
| **Fix is breaking or non-breaking** | Non-breaking                                                         |
| **Enabled by default in .NET 9**    | As information                                                       |

## Cause

> When a nullable struct, e. g. `int?` or `Guid?`, is passed to a call to `ArgumentNullException.ThrowIfNull`, it is boxed to an object, causing a performance penalty.

Reviewer: ```suggestion
When a nullable struct, for example, `int?` or `Guid?`, is passed to `ArgumentNullException.ThrowIfNull`, it's boxed to an object, causing a performance penalty.
```

## Rule description

> Checking the `HasValue` property and manually throwing an exception will lead to improved performance over passing a nullable struct to `ArgumentNullException.ThrowIfNull`.

Reviewer: ```suggestion
For improved performance, it's better to check the `HasValue` property and manually throw an exception than to pass a nullable struct to `ArgumentNullException.ThrowIfNull`.
```
